File: py/harddrive.py
from komponen import Komponen

class Harddrive(Komponen):
    def __init__(self, merk: str, nama: str, tipe_drive: str, kapasitas_gb: int):
        super().__init__(merk, nama)
        self.tipe_drive = tipe_drive
        self.kapasitas_gb = kapasitas_gb

    def get_tipe_drive(self):
        return self.tipe_drive

    def get_kapasitas_gb(self):
        return self.kapasitas_gb

File: java/Harddrive.java
public class Harddrive extends Komponen {
    private String tipeDrive;
    private int kapasitasGB;

    public Harddrive(String merk, String nama, String tipeDrive, int kapasitasGB) {
        super(merk, nama);
        this.tipeDrive = tipeDrive;
        this.kapasitasGB = kapasitasGB;
    }

    public String getTipeDrive() {
        return tipeDrive;
    }

    public int getKapasitasGB() {
        return kapasitasGB;
    }

    public void setTipeDrive(String tipeDrive) {
        this.tipeDrive = tipeDrive;
    }

    public void setKapasitasGB(int kapasitasGB) {
        this.kapasitasGB = kapasitasGB;
    }

    @Override
    public String toString() {
        return super.toString() + " | " + tipeDrive + " | " + kapasitasGB + " GB";
    }
}
